tools/create_db.py

```diff
@@ -0,0 +1,153 @@
+
+# adjust to equired output format. ',' = rest of the world ';' for Germany
+# adding \t in front of the delimiter will prevent conversion to date/number in Excel
+FIELD_SEP = "\t;" 
+
+def assemble_db(backup_details, patch_details):
+    csv = [
+        FIELD_SEP.join(HEADINGS)
+    ]
+    for detail in backup_details:
+        train = detail["Train"]
+        train_brand, train_region = train_split(train)
+        detail["Train_Brand"] = train_brand
+        detail["Train_Region"] = train_region
+        pt = {t:d for t, d in patch_details.items() if t.startswith(train)}
+        if not pt:
+            logging.warning(f"No patch for: {train}")
+        elif len(pt) > 1:
+            logging.warning(f"can't match patch from {list(pt.keys())} patches for: {train}")
+        else:
+            ifs, ifs_sha, ifs_patch, offset, ifs_checksum = list(pt.values())[0]
+            detail["SHA1_patch"] = ifs_sha
+            detail["ifs_header_checksum_patch"] = ifs_patch
+            detail["RCC_offset"] = offset
+            detail["ifs_file_name_checksum_patch"] = ifs_checksum
+            #print(ifs_patch)
+        
+        detail["PN_model"], detail["PN_INDEX"], detail["PN_Ident"] = PN1_split(detail["PN1"])
+        
+        row = FIELD_SEP.join([b2s(detail.get(f, "")) for f in HEADINGS])
+        csv.append(row)
+        
+    logging.info(f"database written: {db_file}")
+    db_file.write_text("\n".join(csv))
+
+
+def b2s(b):
+    return str(b, "utf8") if isinstance(b, bytes) else str(b)
+
+
+def train_split(train):
+    hdr, brand, region, *vers = train.upper().split("_")
+    return region, brand
+
+def ifs_name_split(file_name):
+    MU, ifs1, ifs2, ifs3, offset, ifs_checksum = file_name.lower().split("-")
+    return offset, ifs_checksum
+
+def PN1_split(pn1):
+    PN_model = pn1[0:6]
+    PN_Ident = pn1[6:9]
+    try:
+        PN_INDEX = pn1[9]
+    except IndexError:
+        PN_INDEX = ''
+    return PN_model, PN_INDEX, PN_Ident
+
+
+def sha1sum(filename):
+    h  = hashlib.sha1()
+    b  = bytearray(128*1024)
+    mv = memoryview(b)
+    with open(filename, 'rb', buffering=0) as f:
+        while n := f.readinto(mv):
+            h.update(mv[:n])
+    return h.hexdigest()
+
+
+def parse_backups(backups_dir: Path):
+    logging.info(f"Parsing backups in: {backups_dir} ({backups_dir.exists()})")
+    backup_details = list()
+    for eeprom in backups_dir.glob("*/*-EEProm.bin"): # Change to "**/*-EEProm.bin" to enable scan of sub directories
+        backup = eeprom.parent
+        logging.info(f"Backup: {backup.name}")
+        if os.path.getsize(eeprom) != 8192:
+            logging.warning(f"Unexpected eeprom size: {os.path.getsize(eeprom)}")
+        with eeprom.open("rb") as data:
+            details = parse_eeprom(data)
+        for ifs in backup.glob("*-ifs-root-part2*.ifs"):
+            details["ifs_SHA1"] = sha1sum(ifs)
+        for ifs_header in backup.glob("*-ifs-root-part2*.ifs"):
```

Comment:
@harman-f This appears to be scanning through the same glob files as a couple of lines above so could be combined there

Reply:
fixed in next push